packages/client/test/routings.spec.ts

```diff
@@ -0,0 +1,269 @@
+
+    if (routing == null) {
+      throw new Error('ContentRouting not found')
+    }
+
+    const cid = CID.parse('QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn')
+    const peerId = await createEd25519PeerId()
+    const record = await createIpnsRecord(peerId, cid, 0, 1000)
+
+    // load record for the router to fetch
+    await fetch(`${process.env.ECHO_SERVER}/add-ipns/${peerId.toCID().toString()}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/vnd.ipfs.ipns-record'
+      },
+      body: marshalIpnsRecord(record)
+    })
+
+    const key = uint8ArrayConcat([
+      uint8ArrayFromString('/ipns/'),
+      peerId.toBytes()
+    ])
+
+    const value = await routing.get(key)
+    expect(value).to.equalBytes(marshalIpnsRecord(record))
+  })
+
+  it('should not get other records', async () => {
+    const routing = getContentRouting(client)
+
+    if (routing == null) {
+      throw new Error('ContentRouting not found')
+    }
+
+    const peerId = await createEd25519PeerId()
+
+    const key = uint8ArrayConcat([
+      uint8ArrayFromString('/am-unknown-key/'),
+      peerId.toBytes()
+    ])
+
+    await expect(routing.get(key)).to.eventually.be.rejected
+      .with.property('code', 'ERR_NOT_FOUND')
+
+    await expect(getServerCallCount()).to.eventually.equal(0)
+  })
+})
+
+describe('libp2p peer-routing', () => {
+  let client: DelegatedRoutingV1HttpApiClient
+
+  beforeEach(() => {
+    client = createDelegatedRoutingV1HttpApiClient(new URL(serverUrl))
+  })
+
+  afterEach(async () => {
+    if (client != null) {
+      client.stop()
+    }
+  })
+
+  describe('peer routing', () => {
+    it('should provide a peer routing implementation', () => {
+      const routing = getPeerRouting(client)
+
+      expect(routing).to.be.ok()
+    })
+
+    it('should find peer info', async () => {
+      const routing = getPeerRouting(client)
+
+      if (routing == null) {
+        throw new Error('PeerRouting not found')
+      }
+
+      const peerId = await createEd25519PeerId()
+
+      const records = [{
+        Protocols: ['transport-bitswap'],
+        Schema: 'peer',
+        Metadata: 'gBI=',
+        ID: peerId.toString(),
+        Addrs: ['/ip4/41.41.41.41/tcp/1234']
+      }]
+
+      // load peer for the router to fetch
+      await fetch(`${process.env.ECHO_SERVER}/add-peers/${peerId.toCID().toString()}`, {
+        method: 'POST',
+        body: records.map(prov => JSON.stringify(prov)).join('\n')
+      })
+
+      const peerInfo = await routing.findPeer(peerId)
+
+      expect(peerInfo.id.toString()).to.equal(records[0].ID)
+      expect(peerInfo.multiaddrs.map(ma => ma.toString())).to.deep.equal(records[0].Addrs)
+    })
+
+    it('should not get closest peers', async () => {
+      const routing = getPeerRouting(client)
+
+      if (routing == null) {
+        throw new Error('PeerRouting not found')
+      }
+
+      await drain(routing.getClosestPeers(Uint8Array.from([0, 1, 2, 3, 4])))
```

Comment:
the assertion here is not obvious. it seems like it should be either 
```suggestion
      expect(async () => await drain(routing.getClosestPeers(Uint8Array.from([0, 1, 2, 3, 4])))).toThrow()
      // or
      expect(async () => await drain(routing.getClosestPeers(Uint8Array.from([0, 1, 2, 3, 4])))).to.haveLength(0)
```

Reply:
I've added an assertion.

Note that you have to use the `eventually` keyword to make the kind of assertions you're trying to make above.  See the [chai-as-promised](https://www.chaijs.com/plugins/chai-as-promised/) docs:

```js
// will not work because it'll try to assert that the promise returned from the
// async function has a 'length' property of 0
expect(async () => await drain(routing.getClosestPeers(...))).to.haveLength(0)

// instead:
await expect(drain(routing.getClosestPeers(...))).to.eventually.have.lengthOf(0)
```